neural_modelling/src/delay_extension/delay_extension.c

```diff
@@ -393,21 +437,106 @@ static inline void spike_process(void) {
             n_packets_dropped_due_to_invalid_key += 1;
             log_debug("Invalid spike key 0x%08x", s);
         }
+        state = spin1_int_disable();
     }
 
-    // reactivate interrupts as critical section complete
+    spike_processing = false;
     spin1_mode_restore(state);
 }
 
+static void user_callback(UNUSED uint unused0, UNUSED uint unused1) {
+    spike_process();
+}
+
+static void background_callback(uint local_time, UNUSED uint timer_count) {
+    // reset the tdma for this next cycle.
```

Comment:
Do we want to capitalise TDMA in comments?  Or have something like tDMA instead?

Reply:
I have made them all TDMA in this file